### PR TITLE
Testing: Improve the `@testing-library/react` helper

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

In a [recent review](https://github.com/Automattic/wp-calypso/pull/62920#discussion_r854540445), @noahtallen observed that it wasn't always clear as to whether to import `@testing-library/react` from our test helper, or directly from the package.   Later in a discussion with @flootr, he brought up that this could bring confusion, and it wasn't immediately clear as to why we're importing from our test helper in the first place. 

This PR improves on those points by:

* Renaming our test helper renderer to `renderWithProvider` in order to mirror its real purpose and intention on top of the original `@testing-library/react` rendering function.
* No longer re-export `@testing-library/react` from our test helper, but rather import from it directly.
* Minor: moving the `@testing-library/react` test helper outside of the `config` directory because it was incorrectly put there; this helper has nothing to do with the Calypso config mock helper that lives there.

Being explicit about what we import makes it more obvious as to what we're doing and why we're doing it. 

#### Testing instructions

Verify all tests still pass.

### CODE REVIEW

**Review by noahtallen — APPROVED** [review]
I like it! Will we recommend that folks use `renderWithProvider` for all tests?

File: client/test-helpers/testing-library/index.js [hunk 1]
```diff
@@ -4,7 +4,10 @@ import { applyMiddleware, createStore } from 'redux';
 import thunkMiddleware from 'redux-thunk';
 import initialReducer from 'calypso/state/reducer';
 
-const render = ( ui, { initialState, store, reducers, ...renderOptions } = {} ) => {
+export const renderWithProvider = (
```
Comment (flootr) [inline client/test-helpers/testing-library/index.js:7]: Shall we name it `renderWithReduxProvider` or are we going to add more providers as we need it to that function?

**Comment by flootr** [comment]
Merging despite my [comment](https://github.com/Automattic/wp-calypso/pull/62948#discussion_r856018335), we can follow up if necessary.

File: client/test-helpers/testing-library/index.js (hunk 1 above)
Comment (tyxla) [reply client/test-helpers/testing-library/index.js:7]: Yep, those were my thoughts as well - we can either rename it or add more providers if needed in the future.

**Comment by tyxla** [comment]
> I like it! Will we recommend that folks use `renderWithProvider` for all tests?

I think being specific about when we need a component to render with the providers can be useful.

**Comment by tyxla** [comment]
Thanks for landing it @flootr 🙌

**Pull request merged**